### PR TITLE
pipeline show add --dot option

### PR DESCRIPTION
**Pull request by pared**
Please take closer look on add_argument, I am not quite sure if i got this right.

### CODE REVIEW

File: dvc/cli.py [hunk 1]
```diff
@@ -787,6 +787,11 @@ def parse_args(argv=None):
                         action='store_true',
                         default=False,
                         help='Output DAG as ASCII.')
+    pipeline_show_parser.add_argument(
+                        '--dot',
+                        dest='filename',
```
Comment (efiop) [inline dvc/cli.py]: You don't need the `dest` argument. Without it by default it is going to be `dest='dot'` which a more suitable name for this.

File: dvc/command/pipeline.py [hunk 2]
```diff
@@ -73,13 +73,53 @@ def _show_ascii(self, target, commands, outs):
         d = Dagascii(nodes, edges)
         d.draw()
 
+    def __write_dot(self, target, commands, outs, filename):
```
Comment (efiop) [inline dvc/command/pipeline.py]: There is a lot of common code with `_show_ascii`. Could you please extract it into a `_build_graph` method that is going to be called in both `_show_ascii` and `_write_dot` and is going to return `edges, nodes` pair or something like that?

**Comment by efiop** [comment]
Ah, one more thing. Tests seem to fail mainly because there is no `pydot` installed. You would have to specify it in the `requirements.txt` in order to make tests work.

**Comment by pared** [comment]
flake8 fails again and again on line that seems ok to me, my own installation does not show any problems, is there some special config for flake8 on travis?

**Comment by efiop** [comment]
Looks good! Merged. Thank you!

**Pull request merged**